api/src/pcapi/routes/public/individual_offers/v1/serialization.py

```diff
@@ -194,8 +194,9 @@ def compute_category_fields_model(
     """
     specific_fields: dict[typing.Any, typing.Any] = {}
     for field_name, model_field in ExtraDataModel.__fields__.items():
-        if field_name in subcategory.conditional_fields:
-            is_required = field_name in offers_validation.OFFER_EXTRA_DATA_MANDATORY_FIELDS and method == Method.create
+        conditional_field = subcategories.ConditionalFieldEnum.from_string(field_name).value
+        if conditional_field in subcategory.conditional_fields:
```

Comment:
petite parenthèse : ici on pourrait juste itérer sur `subcategory.conditional_fields` sans avoir à itérer sur `ExtraDataModel.__fields__.items()`